src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -1062,6 +1063,16 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         }
     }
 
+    private fun checkImplDynAutoTrait(holder: RsAnnotationHolder, impl: RsImplItem) {
+        if (impl.traitRef != null) return
+        val typeRef = impl.typeReference ?: return
+        val normType = typeRef.normType
+        if (normType !is TyTraitObject) return
+        if (normType.baseTrait?.isAuto == true) {
```

Comment:
It feels like `baseTrait` behaves wrongly since in this case
```rust
impl dyn Sync {}
```
the compiler emits `E0785`, but the plugin currently emits E1106 (it shouln't), and in this case:
```rust
impl dyn Sync + Send {}
```
the compiler emits `E0785`, but the plugin currently emits nothing. 
I think that `baseTrait` must return `null` if all traits are `auto`.
In this case this condition could be rewritten like 
```suggestion
        if (normType.baseTrait == null) {
```